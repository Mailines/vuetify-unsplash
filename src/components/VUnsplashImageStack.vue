<template>
  <v-card
    class="elevation-0 overflow-y-auto"
    :height="imageHolderHeight + 'px'"
  >
    <v-layout class="mx-6 d-flex justify-center align-center pt-4">
      <stack
        :column-min-width="300"
        :gutter-width="15"
        :gutter-height="15"
        monitor-images-loaded
      >
        <stack-item
          v-for="(image, i) in items"
          class="stack-item-vunsplash d-flex justify-center"
          :key="i"
        >
          <div>
            <v-badge
              class="badge"
              :color="accentColor"
              bordered
              overlap
              :value="image.selected"
            >
              <template v-slot:badge>
                <v-icon
                  class="font-weight-bold"
                  :color="isWhiteTextOnAccent ? '#FFFFFF' : '#000000'"
                  >mdi-check</v-icon
                >
              </template>
              <v-card
                class="pa-0 elevation-0 img-card-vunsplash rounded-lg"
                flat
                color="#00000000"
                @click.native="changeImageSelection(image)"
              >
                <img
                  :style="[
                    image.selected
                      ? {
                          'border-color': accentColor,
                          'border-width': '3px !important'
                        }
                      : {
                          'border-color': '#00000000',
                          'border-width': '0px !important'
                        }
                  ]"
                  class="img-vunsplash rounded-lg"
                  :src="image.urls.small"
                  :alt="image.description"
                  style="width: 100%; height: auto; border: 3px solid #00000000 !important; padding: 0;"
                />
                <div
                  class="overlay-vunsplash rounded-lg"
                  :style="
                    image.selected
                      ? {
                          opacity: '0.5',
                          'background-color': accentColor
                        }
                      : { opacity: '0' }
                  "
                ></div>
              </v-card>
            </v-badge>
          </div>
        </stack-item>
      </stack>
    </v-layout>
  </v-card>
</template>
<script>
import { Stack, StackItem } from "vue-stack-grid";
export default {
  props: {
    imageHolderHeight: {
      type: Number,
      default: window.innerHeight - 370,
      required: true
    },
    items: {
      type: Array,
      required: true
    },
    accentColor: {
      type: String,
      required: true
    },
    isWhiteTextOnAccent: {
      type: Boolean,
      required: true
    }
  },
  components: {
    Stack,
    StackItem
  },
  methods: {
    changeImageSelection(image) {
      this.$emit("changeImageSelection", image);
    }
  }
};
</script>
<style scoped>
.stack-item-vunsplash {
  transition: transform 300ms;
}
.img-card-vunsplash::before {
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  background-color: rgba(0, 0, 0, 0);
  outline: none;
  box-shadow: none !important;
}
.img-vunsplash {
  width: 100%;
  height: auto;
  margin: 0;
  transition: border-width 0.15s ease-out;
}
.overlay-vunsplash {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  height: 100%;
  width: 100%;
  opacity: 0;
  margin: 0;
  padding: 0;
  transition: 0.5s ease;
}
</style>
