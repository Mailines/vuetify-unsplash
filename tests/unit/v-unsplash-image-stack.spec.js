import VUnsplashImageStack from "../../src/components/VUnsplashImageStack";
import { mount, createLocalVue } from "@vue/test-utils";
import Vuetify from "vuetify";

describe("VUnsplashImageStack.vue", () => {
  let localVue;
  let wrapper;
  let vuetify;

  beforeEach(() => {
    localVue = createLocalVue();
    vuetify = new Vuetify();
    wrapper = mount(VUnsplashImageStack, {
      localVue,
      vuetify,
      propsData: {
        imageHolderHeight: 700,
        accentColor: "#1976d2",
        isWhiteTextOnAccent: true,
        items: [
          {
            id: "123456",
            selected: true,
            description: "test image 1",
            urls: {
              thumb:
                "https://www.google.com/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png",
              small:
                "https://www.google.com/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png"
            }
          },
          {
            id: "654321",
            selected: false,
            description: "test image 2",
            urls: {
              thumb:
                "https://www.google.com/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png",
              small:
                "https://www.google.com/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png"
            }
          }
        ]
      }
    });
  });
  it("renders and match snapshot", async () => {
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("emits  changeImageSelection event with image when calling  changeImageSelection(image)", async () => {
    const image = {
      id: "654321",
      selected: false,
      description: "test image 2",
      urls: {
        thumb:
          "https://www.google.com/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png",
        small:
          "https://www.google.com/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png"
      }
    };
    wrapper.vm.changeImageSelection(image);
    expect(wrapper.emitted().changeImageSelection[0][0]).toBe(image);
  });
});
