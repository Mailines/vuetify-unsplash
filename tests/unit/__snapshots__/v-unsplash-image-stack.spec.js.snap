// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VUnsplashImageStack.vue renders and match snapshot 1`] = `
<div class="elevation-0 overflow-y-auto v-card v-sheet theme--light" style="height: 700px;">
  <div class="layout mx-6 d-flex justify-center align-center pt-4">
    <div class="vsg-container" style="height: 30px;">
      <div class="vsg-stack-item stack-item-vunsplash d-flex justify-center" style="width: 0px; transform: translate(0px, 0px);">
        <div><span class="v-badge badge v-badge--bordered v-badge--overlap theme--light"> <div class="pa-0 elevation-0 img-card-vunsplash rounded-lg v-card v-card--flat v-sheet theme--light" style="background-color: rgba(0, 0, 0, 0); border-color: #00000000;"><img src="https://www.google.com/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png" alt="test image 1" class="img-vunsplash rounded-lg" style="width: 100%; height: auto; border: 3px solid #00000000 !important; padding: 0px; border-color: #1976d2; border-width: 3px;"> <div class="overlay-vunsplash rounded-lg" style="opacity: 0.5; background-color: rgb(25, 118, 210);"></div></div><span class="v-badge__wrapper"><transition-stub name="scale-rotate-transition"><span aria-atomic="true" aria-label="Badge" aria-live="polite" role="status" class="v-badge__badge" style="bottom: calc(100% - 12px); left: calc(100% - 12px); background-color: rgb(25, 118, 210); border-color: #1976d2;"><i aria-hidden="true" class="v-icon notranslate font-weight-bold mdi mdi-check theme--light" style="color: rgb(255, 255, 255); caret-color: #FFFFFF;"></i></span></transition-stub></span></span></div>
      </div>
      <div class="vsg-stack-item stack-item-vunsplash d-flex justify-center" style="width: 0px; transform: translate(0px, 15px);">
        <div><span class="v-badge badge v-badge--bordered v-badge--overlap theme--light"> <div class="pa-0 elevation-0 img-card-vunsplash rounded-lg v-card v-card--flat v-sheet theme--light" style="background-color: rgba(0, 0, 0, 0); border-color: #00000000;"><img src="https://www.google.com/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png" alt="test image 2" class="img-vunsplash rounded-lg" style="width: 100%; height: auto; border: 3px solid #00000000 !important; padding: 0px; border-color: #00000000; border-width: 0px;"> <div class="overlay-vunsplash rounded-lg" style="opacity: 0;"></div></div><span class="v-badge__wrapper"><transition-stub name="scale-rotate-transition"><span aria-atomic="true" aria-label="Badge" aria-live="polite" role="status" class="v-badge__badge" style="bottom: calc(100% - 12px); left: calc(100% - 12px); background-color: rgb(25, 118, 210); border-color: #1976d2; display: none;"><i aria-hidden="true" class="v-icon notranslate font-weight-bold mdi mdi-check theme--light" style="color: rgb(255, 255, 255); caret-color: #FFFFFF;"></i></span></transition-stub></span></span></div>
      </div>
    </div>
  </div>
</div>
`;
